domain/src/main/java/org/oppia/domain/oppialogger/exceptions/ExceptionsController.kt

```diff
@@ -0,0 +1,150 @@
+package org.oppia.domain.oppialogger.exceptions
+
+import androidx.lifecycle.LiveData
+import org.oppia.app.model.ExceptionLog
+import org.oppia.app.model.OppiaExceptionLogs
+import org.oppia.data.persistence.PersistentCacheStore
+import org.oppia.domain.oppialogger.ExceptionLogStorageCacheSize
+import org.oppia.util.data.AsyncResult
+import org.oppia.util.data.DataProviders
+import org.oppia.util.logging.ConsoleLogger
+import org.oppia.util.logging.ExceptionLogger
+import org.oppia.util.networking.NetworkConnectionUtil
+import javax.inject.Inject
+
+/** Controller for handling exception logging. */
+class ExceptionsController @Inject constructor(
+  private val exceptionLogger: ExceptionLogger,
+  private val dataProviders: DataProviders,
+  cacheStoreFactory: PersistentCacheStore.Factory,
+  private val consoleLogger: ConsoleLogger,
+  private val networkConnectionUtil: NetworkConnectionUtil,
+  @ExceptionLogStorageCacheSize private val exceptionLogStorageCacheSize: Int
+) {
+  private val exceptionLogStore =
+    cacheStoreFactory.create("exception_logs", OppiaExceptionLogs.getDefaultInstance())
+
+  /** Logs a NON-FATAL exception. */
+  fun logNonFatalException(exception: Exception, timestamp: Long) {
+    uploadOrCacheExceptionLog(exception, timestamp, ExceptionLog.ExceptionType.NON_FATAL)
```

Comment:
Can we import ExceptionLog.ExceptionType and directly use ExceptionType.NON_FATAL/FATAL

Reply:
Done.

Reply:
Doesn't appear done--this still looks like ``ExceptionLog.ExceptionType.NON_FATAL``.

Reply:
Oh my bad, I did it in the test file. Done now.